### PR TITLE
Fix generated docs after release

### PR DESCRIPTION
**Pull request by matthchr**
* Docs now indicate the new resources are released.
* Updated generator to correctly exit with a nonzero exit code if an error was returned.
* Resolved an issue with the generator where not consuming the deprecated or prerelease fragments for documentation generation was fatal.

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

**Review by super-harsh — APPROVED** [review]

File: v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go [hunk 1]
```diff
@@ -333,6 +333,8 @@ func (report *ResourceVersionsReport) writeGroupSections(
 	if err != nil {
 		return errors.Wrapf(err, "writing prerelease resources for group %s", group)
 	}
+	// Prerelease may have had no usage, but consume its fragment anyway
+	report.findFragment("prerelease")
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go:337]: Add a blank line before this to separate it from the error handling immediately above. #nit

File: v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go [hunk 2]
```diff
@@ -353,6 +355,7 @@ func (report *ResourceVersionsReport) writeGroupSections(
 	if err != nil {
 		return errors.Wrapf(err, "writing deprecated resources for group %s", group)
 	}
+	report.findFragment("deprecated")
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go:358]:
```suggestion

	// 'deprecated' may have had no usage, but consume the fragment anyway
	report.findFragment("deprecated")
```

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go (hunk 2 above)
Comment (matthchr) [reply v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go:358]: Fixed

**Pull request merged**